src/arch/arm/32/kernel/vspace.c

```diff
@@ -1122,52 +1132,24 @@ pde_t *pageTableMapped(asid_t asid, vptr_t vaddr, pte_t *pt)
 
 static void invalidateASID(asid_t asid)
 {
-    asid_pool_t *asidPool;
-    pde_t *pd;
-
-    asidPool = armKSASIDTable[asid >> asidLowBits];
-    assert(asidPool);
+    asid_map_t *asid_map;
 
-    asid_map_t asid_map = asidPool->array[asid & MASK(asidLowBits)];
-    assert(asid_map_get_type(asid_map) == asid_map_asid_map_vspace);
+    asid_map = findMapRefForASID(asid);
+    assert(asid_map_get_type(*asid_map) == asid_map_asid_map_vspace);
 
-    pd = (pde_t *)asid_map_asid_map_vspace_get_vspace_root(asid_map);
-
-    pd[PD_ASID_SLOT] = pde_pde_invalid_new(0, false);
-}
-
-static pde_t PURE loadHWASID(asid_t asid)
-{
-    asid_pool_t *asidPool;
-    pde_t *pd;
-
-    asidPool = armKSASIDTable[asid >> asidLowBits];
-    assert(asidPool);
-
-    asid_map_t asid_map = asidPool->array[asid & MASK(asidLowBits)];
-    assert(asid_map_get_type(asid_map) == asid_map_asid_map_vspace);
-
-    pd = (pde_t *)asid_map_asid_map_vspace_get_vspace_root(asid_map);
-
-    return pd[PD_ASID_SLOT];
+    asid_map_asid_map_vspace_ptr_set_hw_asid(asid_map, 0);
+    asid_map_asid_map_vspace_ptr_set_hw_asid_valid(asid_map, false);
 }
 
 static void storeHWASID(asid_t asid, hw_asid_t hw_asid)
 {
-    asid_pool_t *asidPool;
-    pde_t *pd;
-
-    asidPool = armKSASIDTable[asid >> asidLowBits];
-    assert(asidPool);
-
-    asid_map_t asid_map = asidPool->array[asid & MASK(asidLowBits)];
-    assert(asid_map_get_type(asid_map) == asid_map_asid_map_vspace);
+    asid_map_t *asid_map;
 
-    pd = (pde_t *)asid_map_asid_map_vspace_get_vspace_root(asid_map);
+    asid_map = findMapRefForASID(asid);
+    assert(asid_map_get_type(*asid_map) == asid_map_asid_map_vspace);
 
-    /* Store HW ASID in the last entry
-       Masquerade as an invalid PDE */
-    pd[PD_ASID_SLOT] = pde_pde_invalid_new(hw_asid, true);
```

Comment:
This should mean that the MODE_RESERVED region for this region of untyped can be removed.